src/cargo/util/command_prelude.rs

```diff
@@ -986,6 +1001,44 @@ pub fn root_manifest(manifest_path: Option<&Path>, gctx: &GlobalContext) -> Carg
     }
 }
 
+pub fn lockfile_path(
+    lockfile_path: Option<&Path>,
+    gctx: &GlobalContext,
+) -> CargoResult<Option<PathBuf>> {
+    let Some(lockfile_path) = lockfile_path else {
+        return Ok(None);
+    };
+
+    gctx.cli_unstable()
+        .fail_if_stable_opt("--lockfile-path", 5707)?;
+
+    let path = gctx.cwd().join(lockfile_path);
+
+    if !path.ends_with(LOCKFILE_NAME) {
+        bail!("the lockfile-path must be a path to a {LOCKFILE_NAME} file")
+    }
+    if path.is_dir() {
+        bail!(
+            "lockfile path `{}` is a directory but expected a file",
+            lockfile_path.display()
+        )
+    }
+    if !path.exists() {
+        // Root case should already be covered above
+        let parent_path = lockfile_path.parent().expect("lockfile path can't be root");
+
+        // Fail early if parent path existence can't be verified (e.g. due to permissions)
+        parent_path.try_exists().with_context(|| {
+            format!(
+                "failed to fetch lock file's parent path metadata {}",
+                parent_path.display()
+            )
+        })?;
+    }
```

Comment:
I've not been following the review too closely lately but it looks like this errors if the parent directory is not present

From #5707

> Main question is how this should work if the lockfile or a parent directory is not present
>
> > I would lean towards "create them" rather than "error"

Could you call out in the PR description why you made that behavior decision?

Reply:
Hi Ed, we are creating parent directory in [write_pkg_lockfile](https://github.com/rust-lang/cargo/blob/76d75059091923fc00a4749090d93151d1d418f7/src/cargo/ops/lockfile.rs#L87-L89). I added this check so we can fail early if we can't check for parent path and know it can't be created (e.g. loop symlinks).
But now that I think of it, if we have write only permission on a directory (`-w-`), then this check will fail when we still can create directory. Let me actually remove it